docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+<!-- TODO: point to the direct db find method for the alternate way to project results. -->
+
+### `relationships()`
+
+```
+query.relationships(false)
+
+query.relationships(['_author'])
+```
+
+By default, [relationship](/guide/relationships.md) data are fetched for queried documents. The `relationships` builder can prevent or limit this.
+
+Pass `false` to the builder to not retrieve any relationship data. You may also pass an array of relationship names, which will fetch only those relationships and those intermediate to them (using dot notation).
+
+### `sort()`
+
+```
+query.sort({ updatedAt: -1, title: 1 })
+```
+
+The `sort` builder sets the sorting order for returned documents. If not set, the default is used, from the [`defaultSort` builder](#defaultsort) or module configuration. The sort argument is an object like `{ updatedAt: 1 }`, including a field name and `1` for ascending or `-1` for descending.  See the [MongoDB `sort` method documentation](https://docs.mongodb.com/manual/reference/method/cursor.sort/) for other options.
+
+`false` can be passed as an option to use *no sort definition at all*. This can be helpful when using MongoDB operators like `$near`, which already sort.
+
+If this method is never called or the argument is `undefined`, a case-insensitive sort on the title is normally the default. If `search()` has been called, then a sort by search result quality is the default.
+
+If the query sorts on a field that is defined in a doc type's schema with the `sortify: true` option, then this query builder will automatically substitute a "sortified" version of the field: case-insensitive and ignoring extra whitespace and punctuation. This provides a more natural sort than MongoDB normally does.
+
+### `skip()`
+
+```
+query.skip(10)
+```
+
+The `skip` builder accepts a number as an argument, then skips that number of documents in a query's results. This affects `toArray` and `toObject`. It does not affect `toDistinct` or `toMongo`.
+
+### `search()`
+
+```
+query.search('tree')
+```
+
+The `search` builder limits results to those matching the string passed as an argument. Search is implemented using MongoDB's `$text` operator and a full text index.
+
+If this query builder is set, the `sort` query builder will default to sorting by search quality. This is important because the worst of the full-text search matches will be of very poor quality.
+
+### `type()`
+
+```
+query.type('product')
+```
+
+The `type` builder can be used to limit a query to only one particular doc type. Pass the doc type name (string) as an argument.
+
+Generally you don't want to call this method directly. It will be better to include the doc type name as the `type` criteria in the original `find()` method arguments.
```

Comment:
If you are invoking the `find` method of a particular piece type, then this is invoked for you and there is no need to worry about documents of the wrong type being returned. It is primarily useful in implementing special case "virtual piece types" like Apostrophe's `submitted-draft` module, which displays pieces of many types by extending its `find` method to pass `null` to this builder.